engine/src/main/java/org/hibernate/validator/internal/xml/ConstrainedExecutableBuilder.java

```diff
@@ -322,17 +322,14 @@ private CascadingTypeParameter parseReturnValueType(ReturnValueType returnValueT
 
 	private CascadingTypeParameter getCascadingMetaDataForReturnValue(Map<TypeVariable<?>, CascadingTypeParameter> containerElementTypesCascadingMetaData, Executable executable,
 			ReturnValueType returnValueType, String defaultPackage) {
-		boolean isArray = executable instanceof Method && ( (Method) executable ).getReturnType().isArray();
 		Type type = ReflectionHelper.typeOf( executable );
 		boolean isCascaded = returnValueType.getValid() != null;
 		Map<Class<?>, Class<?>> groupConversions = groupConversionBuilder.buildGroupConversionMap(
 				returnValueType.getConvertGroup(),
 				defaultPackage
 		);
 
-		return isArray
-				? CascadingTypeParameter.arrayElement( type, isCascaded, containerElementTypesCascadingMetaData, groupConversions )
```

Comment:
It's not removed because we still use it in the code I kept if we want to revive array support. It makes sense for the array elements but here we are talking about the whole array so it should be an annotated object.